### DISCUSSION TITLE
JSON content validator

### DISCUSSION THREAD

**Discussion by estarossa0** [ueberdosis/tiptap#2215] [General Questions] [answered]
Hi, I'm using json format to save editor content in my db, and I would like to know is there is a proper way to validate the received data from client as a valid editor content.  thank you 🙏

**Comment by omarkhairy21** [comment]
i have the same question

**Comment by christianheine** [comment, marked as answer]
You can retrieve the underlying ProseMirror node and [check it](https://prosemirror.net/docs/ref/#model.Node.check) for consistency. The `check` method will throw on invalid schemas. Below implementation was inspired by the [generateHTML function](https://tiptap.dev/api/utilities/html) - which is using a similar approach.

```typescript 

import { Node } from "prosemirror-model";
import { Extensions, getSchema, JSONContent } from "@tiptap/core";

const validateSchema = (doc: JSONContent, extensions: Extensions): boolean => {
  try {
    const schema = getSchema(extensions);
    const contentNode = Node.fromJSON(schema, doc);
    contentNode.check();
    return true;
  } catch (e) {
    return false;
  }
};
```

**Reply by CapitaineToinon** [reply]
I think this is outdated? I'm trying to do something similar with tiptap 2+ to valid the schema server side

**Reply by nabramow** [reply]
@CapitaineToinon Did you ever solve this? I am trying to do the same thing (validate the tiptap schema server side).

**Reply by jasonhao518** [reply]
I'm using chatgpt to generate json format response, need to let chatgpt know the schema. but cannot find schema definition anywhere.

**Reply by mathers101** [reply]
@nabramow I was having same issue with this code not working server side, ended up just writing the schema from scratch as described on the following page (I guess this only works if you have a very specific schema you'll consistently be using): https://prosemirror.net/examples/schema/

After doing it this way I am able to use the same code, but without using the getSchema function, and it works as intended
